wen-restart/src/last_voted_fork_slots_aggregate.rs

```diff
@@ -141,70 +140,46 @@ impl LastVotedForkSlotsAggregate {
     pub(crate) fn aggregate(
         &mut self,
         new_slots: RestartLastVotedForkSlots,
-    ) -> Option<LastVotedForkSlotsRecord> {
+    ) -> LastVotedForkSlotsAggregateResult {
         let from = &new_slots.from;
         if from == &self.my_pubkey {
-            return None;
+            return LastVotedForkSlotsAggregateResult::AlreadyExists;
         }
         let root_slot = self.root_bank.slot();
         let new_slots_vec = new_slots.to_slots(root_slot);
         if new_slots_vec.is_empty() {
-            return None;
+            // This could be a validator that has super old vote, we still want to
+            // count it in active peers though because it will switch to agreed upon
+            // heaviest fork later.
```

Comment:
might be good to add a comment that `to_slots` above has filtered out everything `<= root_slot`

Reply:
Done